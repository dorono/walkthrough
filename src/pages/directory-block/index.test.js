import React from 'react';
import {shallow} from 'enzyme';
import DirectoryBlockPage from './index';

const defaultProps = {
    data: {
        ablock: {
            hash: 'hash1',
            entries: {
                count: 4,
            },
        },
        ecblock: {
            hash: 'hash2',
            entries: {
                count: 6,
            },
        },
        eblocks: [{
            keymr: 'keymr',
            chain: {
                chain_id: 'chain1',
            },
            entries: {
                count: 23,
            },
        }],
        fblock: {
            hash: 'hash3',
            transactions: {
                count: 4,
            },
        },
    },
    location: {},
};

jest.mock('../../utils/date', () => {
    return {
        currentTimezone: () => '11/11/11',
        formatDateLong: () => '11111111',
    };
});

describe('DirectoryBlock', () => {
    it('should render without errors', () => {
        shallow(<DirectoryBlockPage {...defaultProps} />);
    });

    it('should match snapshot', () => {
        expect(shallow(<DirectoryBlockPage {...defaultProps} />)).toMatchSnapshot();
    });
});
