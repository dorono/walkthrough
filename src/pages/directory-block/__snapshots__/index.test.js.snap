// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DirectoryBlock should match snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
